test/integ/quote.test.ts

```diff
@@ -1,808 +1,1078 @@
-        status,
-      } = response
-
-      expect(status).toBe(200)
-      expect(parseFloat(quoteDecimals)).toBeGreaterThan(90)
-
-      if (type == 'exactIn') {
-        expect(parseFloat(quoteGasAdjustedDecimals)).toBeLessThanOrEqual(parseFloat(quoteDecimals))
-      } else {
-        expect(parseFloat(quoteGasAdjustedDecimals)).toBeGreaterThanOrEqual(parseFloat(quoteDecimals))
-      }
-
-      expect(parseFloat(quoteDecimals)).toBeLessThan(110)
-    })
-
-    test('erc20 -> erc20 one by address one by symbol', async () => {
-      const quoteReq: QuoteQueryParams = {
-        tokenInAddress: '0x6B175474E89094C44Da98b954EedeAC495271d0F',
-        tokenInChainId: 1,
-        tokenOutAddress: 'USDC',
-        tokenOutChainId: 1,
-        amount: await getAmount(type, 'DAI', 'USDC', '100'),
-        type,
-        recipient: '0xAb5801a7D398351b8bE11C439e05C5B3259aeC9B',
-        slippageTolerance: '5',
-        deadline: '360',
-        algorithm,
-      }
-
-      const queryParams = qs.stringify(quoteReq)
-
-      const response: AxiosResponse<QuoteResponse> = await axios.get<QuoteResponse>(`${API}?${queryParams}`)
-      const {
-        data: { quoteDecimals, quoteGasAdjustedDecimals },
-        status,
-      } = response
-
-      expect(status).toBe(200)
-      expect(parseFloat(quoteDecimals)).toBeGreaterThan(90)
-
-      if (type == 'exactIn') {
-        expect(parseFloat(quoteGasAdjustedDecimals)).toBeLessThanOrEqual(parseFloat(quoteDecimals))
-      } else {
-        expect(parseFloat(quoteGasAdjustedDecimals)).toBeGreaterThanOrEqual(parseFloat(quoteDecimals))
-      }
-
-      expect(parseFloat(quoteDecimals)).toBeLessThan(110)
-    })
+const callAndExpectFail = async (quoteReq: Partial<QuoteQueryParams>, resp: { status: number; data: any }) => {
+  const queryParams = qs.stringify(quoteReq)
+  try {
+    await axios.get<QuoteResponse>(`${API}?${queryParams}`)
+    fail()
+  } catch (err) {
+    expect(err.response).to.containSubset(resp)
+  }
+}
+
+const checkQuoteToken = (
+  before: CurrencyAmount<Currency>,
+  after: CurrencyAmount<Currency>,
+  tokensQuoted: CurrencyAmount<Currency>
+) => {
+  // Check which is bigger to support exactIn and exactOut
+  const tokensSwapped = after.greaterThan(before) ? after.subtract(before) : before.subtract(after)
+
+  const tokensDiff = tokensQuoted.greaterThan(tokensSwapped)
+    ? tokensQuoted.subtract(tokensSwapped)
+    : tokensSwapped.subtract(tokensQuoted)
+  const percentDiff = tokensDiff.asFraction.divide(tokensQuoted.asFraction)
+  /* console.log(
+    `5pc: ${new Fraction(5, 100).toFixed(10)} tokens diff: ${tokensDiff.toExact()} percent diff: ${percentDiff.toFixed(
+      5
+    )} ${percentDiff.quotient} ${percentDiff.remainder.toFixed(5)} (${percentDiff.numerator}/${
+      percentDiff.denominator
+    }), quote: ${tokensQuoted.toExact()}, rceived: ${tokensSwapped.toExact()}`
+  ) */
+  expect(percentDiff.lessThan(new Fraction(5, 100))).to.be.true
```

Comment:
where does this figure come from??

Reply:
All the tests specify 5% for slippage. Refactored this to a constant